src/Microsoft.OData.Client/ALinq/ResourceBinder.cs

```diff
@@ -46,6 +44,11 @@ internal class ResourceBinder : DataServiceALinqExpressionVisitor
         /// </summary>
         private readonly DataServiceContext context;
 
+        /// <summary>
+        /// If true, a Where clause that compares only the key property, will generate a $filter query option.
+        /// </summary>
+        private static bool keyComparisonGeneratesFilterQuery;

```

Comment:
Why is this field static?

Reply:
@habbes It's being used within a static method

Reply:
Resolved by passing the context in the `AnalyzePredicate` method